tests/bsim/bluetooth/ll/conn/src/test_connect2.c

```diff
@@ -209,7 +209,12 @@ static void test_con2_main(void)
 	 * of starting delayed work so we do it here
 	 */
 	while (1) {
-		k_sleep(K_SECONDS(1));
+		if (IS_ENABLED(CONFIG_TEST_CONN_INTERVAL_1MS) ||
+		    IS_ENABLED(CONFIG_BT_CTLR_TX_DEFER)) {
+			k_sleep(K_MSEC(1));
+		} else {
+			k_sleep(K_SECONDS(1));
+		}
```

Comment:
Hmm, this looks kind of odd. I guess it makes sense to send a notification every 1ms when testing with 1ms conn interval, but why are we sending every 1 second if not? What if the interval is 10ms, shouldn't we send every 10ms then? What if the connection interval is 4 seconds?

Shouldn't we just ignore the Kconfig values and just always sleep for the current connection interval?

Reply:
The original (probably the very first) bsim test simulated a peripheral HR service where the notification were 1 second apart. Do not want to change the existing test behavior in this PR.

Only the verification that the number of notifications received in the 3 seconds of simulation has been updated.

Reply:
> Do not want to change the existing test behavior in this PR.

We can create a PR modifying the behavior to use the connection interval first, and then you wouldn't need this change at all ;)